src/main/java/hng_java_boilerplate/user/controller/UserController.java

```diff
@@ -24,4 +25,23 @@ public class UserController {
     public ResponseEntity<?> getUserDetails(@PathVariable String userId) {
         return ResponseEntity.ok(userService.getUserWithDetails(userId));
     }
+
+    @PreAuthorize("hasRole('ROLE_SUPER_ADMIN')")
+    @GetMapping(value = "/members", produces = "application/json")
+    public ResponseEntity<?> getAllMembers(@RequestParam int page, Authentication authentication) {
+        List<MembersResponse> allUsers = userService.getAllUsers(page, authentication);
+        Response<?> response = Response.builder().message("Users List Successfully Fetched").status_code("200").data(allUsers).build();
+        return ResponseEntity.status(HttpStatus.OK).body(response);
+    }
+
+    @GetMapping("/me/{id}")
+    public ResponseEntity<?> getUserById(@PathVariable String id, Authentication authentication) {
+        try {
+            Response<?> response = userService.getUserById(id, authentication);
+            return ResponseEntity.status(response.getStatus_code().equals("200") ? 200 : 401).body(response);
+        } catch (NotFoundException e) {
```

Comment:
You can throw not found in the service class and the handler will pick it up

Reply:
comment reviewed